assets/scripts/dialogs/Geotag/LocationPopup.jsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Popup } from 'react-leaflet'
+import { FormattedMessage } from 'react-intl'
+
+const POPUP_MAX_WIDTH = 300
+const POPUP_OFFSET = [0, -30]
+
+export default class LocationPopup extends React.Component {
+  static propTypes = {
+    position: PropTypes.shape({
+      lat: PropTypes.number,
+      lng: PropTypes.number
+    }),
+    label: PropTypes.string,
+    isEditable: PropTypes.bool,
+    isClearable: PropTypes.bool,
+    handleConfirm: PropTypes.func,
+    handleClear: PropTypes.func
+  }
+
+  static defaultProps = {
+    isEditable: false,
+    isClearable: false
+  }
+
+  renderLocationButton = () => {
+    if (!this.props.isEditable) return null
+
+    return (this.props.isClearable) ? (
+      <button className="geotag-location-button" onClick={this.props.handleClear}>
+        <FormattedMessage id="dialogs.geotag.clear-location" defaultMessage="Clear location" />
+      </button>
+    ) : (
+      <button className="geotag-location-button" onClick={this.props.handleConfirm}>
+        <FormattedMessage id="dialogs.geotag.confirm-location" defaultMessage="Confirm location" />
+      </button>
+    )
+  }
+
+  render () {
+    if (!this.props.position) return null
+
+    return (
+      <Popup
+        position={this.props.position}
+        offset={POPUP_OFFSET}
+        maxWidth={POPUP_MAX_WIDTH}
+        closeButton={false}
+        closeOnClick={false}
+      >
+        <span>{this.props.label}</span>
+        {this.renderLocationButton()}
```

Comment:
Ahh... I'll probably change this to `<p>` or `<div>` wrapping elements then. Block-level elements are easier to adjust for spacing between UI elements. I wanted to avoid `<br/>` elements, which are weird because they're not containers and you have no control over things like margins.